infra/build/functions/request_build/main.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2020 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+"""Cloud function to request builds."""
+import base64
+import logging
+import sys
+import yaml
+
+import google.auth
+from googleapiclient.discovery import build
+from google.cloud import ndb
+
+import build_project
+import build_lib
+
+BASE_PROJECT = 'oss-fuzz-base'
+
+
+# pylint: disable=too-few-public-methods
+class Project(ndb.Model):
+  """Represents an integrated OSS-Fuzz project."""
+  name = ndb.StringProperty()
+  schedule = ndb.StringProperty()
+  project_yaml_contents = ndb.StringProperty()
+  dockerfile_lines = ndb.StringProperty(repeated=True)
+
+
+# pylint: disable=no-member
+def request_build(event, context):
+  """Entry point for cloud function to request builds."""
+  del context  #unused
+  if 'data' in event:
+    project_name = base64.b64decode(event['data']).decode('utf-8')
+  else:
+    logging.error('Project name missing from payload')
+    sys.exit(1)
+  client = ndb.Client()
+  with client.context():
+    query = Project.query(Project.name == project_name)
+    if not query:
```

Comment:
Does this ever actually happen? I've never seen this. I think you need to have the check after line 59, and check the result of `get` instead.